### PR TITLE
Add recursivity and tree to software license type dropdown

### PR DESCRIPTION
**Pull request by wawax**
Make software license type dropdown recursive and tree dropdown

### CODE REVIEW

File: inc/softwarelicensetype.class.php [hunk 1]
```diff
@@ -42,6 +42,7 @@
 /// Class SoftwareLicenseType
 class SoftwareLicenseType extends CommonDropdown {
```
Comment (orthagh) [inline inc/softwarelicensetype.class.php]: CommonTreeDropdown so ?

Comment (wawax) [inline inc/softwarelicensetype.class.php]: yes you're right

File: install/mysql/glpi-9.2-empty.sql [hunk 2]
```diff
@@ -6776,17 +6776,25 @@ CREATE TABLE `glpi_softwarelicenses` (
...
   PRIMARY KEY (`id`),
   KEY `name` (`name`),
   KEY `date_mod` (`date_mod`),
-  KEY `date_creation` (`date_creation`)
+  KEY `date_creation` (`date_creation`),
+  KEY `softwarelicensetypes_id` (`softwarelicensetypes_id`)
 ) ENGINE=MyISAM  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
 
-INSERT INTO `glpi_softwarelicensetypes` VALUES ('1','OEM','',NULL,NULL);
+INSERT INTO `glpi_softwarelicensetypes` VALUES ('1', '0', '1', 'OEM','',NULL,NULL, '0', 'OEM', '0','','');
```
Comment (ddurieux) [inline install/mysql/glpi-9.2-empty.sql]: Like others INSERT into empty SQL, `ancestors_cache` and `sons_cache` must be `NULL` by default, and not empty `''`

Comment (wawax) [inline install/mysql/glpi-9.2-empty.sql]: yes you're right. I've fixed the default values.
thank you David.

File: inc/commondbtm.class.php [hunk 3]
```diff
@@ -3754,6 +3754,7 @@ function deleteByCriteria($crit=array(), $force=0, $history=1) {
          $crit['FIELDS'] = 'id';
          $ok = true;
          foreach ($DB->request($this->getTable(), $crit) as $row) {
+            Toolbox::logDebug("delete ".$row['id']);
```
Comment (orthagh) [inline inc/commondbtm.class.php]: remove logDebug

File: install/update_91_92.php [hunk 4]
```diff
@@ -72,6 +72,21 @@ function update91to92() {
    }
 
    //put you migration script here
+   $migration->addField("glpi_softwarelicensetypes", "softwarelicensetypes_id", "integer");
```
Comment (orthagh) [inline install/update_91_92.php]: add a comment for the topic

**Comment by orthagh** [comment]
still 2 minor issues, i think we can merge after the fixes

**Pull request merged**